src/System.Runtime/tests/System/GCTests.cs

```diff
@@ -75,6 +75,7 @@ public static void Collection_InvalidCollectionMode_ThrowsArgumentOutOfRangeExce
         }
 
         [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Mono)]
```

Comment:
Are there plans to update the mono runtime to have similar semantics?  Seems like a fairly significant discrepancy that WaitForPendingFinalizers doesn't wait for pending finalizers.

Reply:
It's actually two separate issues.

Mono doesn't have precise garbage collection in relation to stack scanning. I believe the tests that require it should be conditioned and the condition should have appropriate name. `GC.WaitForPendingFinalizers` does wait for pending finalizers but due to imprecise stack scanning some objects can stay alive through `GC.Collect`.

On one occasion I was able to trip the garbage collector in a way that it marked the object as unreferenced, but some later scanning pass kept it alive. This is likely a bug and separate issue unlike the imprecise garbage collection which is a deliberate decision.

Reply:
We should have a capability flag for precise GC and disable the test that expected precise GC against that.

Reply:
Done, let me know if you don't like the proposed name